transmission_interface/CMakeLists.txt

```diff
@@ -12,23 +12,24 @@ endif()
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
-find_package(hardware_interface REQUIRED)
-find_package(tinyxml2_vendor REQUIRED)
-find_package(TinyXML2 REQUIRED)
+find_package(rclcpp REQUIRED)
```

Comment:
I vote against `rclcpp` here. From what I see you're only requiring it because of the logging functionality?
What about using `rcutils` logging for this: https://github.com/ros2/rcutils/blob/master/include/rcutils/logging.h

`rclcpp` is a huge dependency and I would like to avoid it as much as possible.

Reply:
I could switch to rcutils. Is this what rclcpp logger is based on?

Another alternative would be to throw std::runtime_error or a custom exception class. It even makes some sense that a parser should not output logging messages but rather pass the error up to the caller either through exception handling or out-arg. It is the caller that should decide if it becomes output to the user or ignored. Thoughts?  What is the policy on using c++ exceptions in ros2?

Reply:
Yes, rclcpp logging is based on rclutils, so I will switch to this for logging.

Reply:
Bence removed the rclcpp dependancy.